dvc/api.py

```diff
@@ -0,0 +1,25 @@
+from dvc.utils.compat import open as _open
+from dvc.repo import Repo
+
+
+def get_url(path, repo_dir=None, remote=None):
+    """Returns an url of `path` in default or specified remote"""
+    repo = Repo(repo_dir)
+    out, = repo.find_outs_by_path(path)
+    # import ipdb; ipdb.set_trace()
+    return str(repo.get_remote(remote).checksum_to_path_info(out.checksum))
+
+
+def open(path, repo_dir=None, remote=None):
+    """Opens a specified resource as a file descriptor"""
+    repo = Repo(repo_dir)
```

Comment:
Same here. Something like
```
repo = Repo(repo_dir)
with repo.open(path) as fobj:
   ...
```
Don't really see a point of making it a separate `dvc/api`, if `Repo` is the API anyways.

Reply:
This was part of initial discussion and we decided to make this separate.

Reply:
It feels like that logic belongs to Repo. Even if we need get_url detached from Repo for convenience, we could simply make it a wrapper around a more generic Repo API. Something like:
```
def open(path, repo_dir=None, remote=None):
    return Repo(repo_dir).open(path, remote=remote)
```
What do you think?

Reply:
This api is going to be publicly released soon with this "model deployment" feature. I'm not sure we are ready to release `Repo` api at that time. So, it makes sense to keep the public part (`dvc.api`) and upcoming full api (`dvc.repo`) separately at least for now.

It might be a good idea to create wrappers in `Repo` if other modules can benefit from this. Otherwise, I'd suggest postponing these activities until we will be ready to polish and release `repo` api.

Reply:
@dmpetrov The approach I'm suggesting above doesn't actually mean that we need to make Repo officially public. I'm just saying that `open`'s logic is using Repo internals and in the current state it looks like it belongs to Repo. The "api" part that we want to release is simply going to use Repo's public methods inside and act as a wrapper

Reply:
I am with @dmpetrov here, create methods in `Repo` when it needed later.

Reply:
One more thought about the Repo API. Today it is not clear what kind of dvc-API need to be released. When we go through use cases we might decide that Repo is not what users need.

Why?
1. Repo class reflects CLI\procedural style where a more functional or OOP style might be a better fit for programmatic API of DVC. For example, AirFlow uses OOP style when the major object is step\stage, not Repo state.
2. Public API will require a lot of syntax sugar that will explode the size of Repo (which is already large class with a number of code smells).
3. We might decide to focus on python functions (not just file\commands) as a basic pipeline entity which is a much higher level of abstraction compared to Repo.

So, I'd not overfocus on external Repo API for now. We should focus on internal code design for developers(for ourselves).